src/test/files/checks/slingquery/SlingQueryImplicitStrategyCheck.java

```diff
@@ -0,0 +1,114 @@
+/*-
+ * #%L
+ * AEM Rules for SonarQube
+ * %%
+ * Copyright (C) 2015 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.statistics.accelerator.foundation.datasource;
+
+import static org.apache.sling.query.SlingQuery.$;
+
+import java.io.IOException;
+import java.util.Optional;
+import javax.servlet.ServletException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceMetadata;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.query.SlingQuery;
+import org.apache.sling.query.api.SearchStrategy;
+import org.apache.tools.ant.taskdefs.optional.Script;
+
+public class SlingQueryImplicitStrategyCheck extends SlingAllMethodsServlet {
```

Comment:
Pretty cool set of tests!

Reply:
@michaltobiasz should receive the praise.